src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateCopy = { ...state };
+  const stateHistory = [];
+
+  for (let index = 0; index < actions.length; index++) {
+    const action = actions[index];
+    const type = action.type;
+    const keys = Object.keys(stateCopy);
+
+    switch (type) {
+      case 'addProperties':
+        stateCopy = { ...stateCopy, ...action.extraData };
+        stateHistory.push({ ...stateCopy });
```

Comment:
[DONT REPEAT YOURSELF] Instead of pushing the `stateCopy` to `stateHistory` in each case, do it once after the `switch` block to avoid repetition.